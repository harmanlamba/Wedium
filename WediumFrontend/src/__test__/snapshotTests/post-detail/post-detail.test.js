import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import moment from 'moment';

import PostDetailInfo from '../../../components/post-detail/post-detail-info';

it('snapshot postDetailInfo default load', () => {
  const posts = [
    {
      articleImageUrl: 'Image URL',
      articleUrl: 'Article URL',
      date: moment.utc().valueOf(),
      description: 'Post Description',
      isFavourited: false,
      isPostLiked: false,
      numberOfLikes: 2,
      postId: 3028,
      postType: 'Culture',
      title: 'Post Title',
      username: 'Bob Bobby',
    },
  ];

  const setRouteLeaveHook = jest.fn();

  let wrapper = shallow(
    <PostDetailInfo.WrappedComponent
      posts={posts}
      post={posts[0]}
      params={{
        router: setRouteLeaveHook,
      }}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
