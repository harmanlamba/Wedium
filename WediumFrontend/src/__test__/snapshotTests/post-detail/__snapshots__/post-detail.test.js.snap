// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot postDetailInfo default load 1`] = `
<div>
  <div
    className="makeStyles-buttonGroup-6"
  >
    <div
      className="makeStyles-buttonGroupFlex-7"
    >
      <WithStyles(Connect(PostLike))
        isPostLiked={false}
        numberOfLikes={2}
        postId={3028}
      />
      <WithStyles(Connect(withRouter(PostFavourite)))
        isFavourited={false}
        postId={3028}
      />
      <WithStyles(CommentButton) />
    </div>
  </div>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-postHeader-2"
  >
    <WithStyles(ForwardRef(Card))
      className="makeStyles-root-1"
      elevation={0}
    >
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          variant="caption"
        >
          <span
            className="makeStyles-postType-3"
          >
            Culture
          </span>
          <WithStyles(ForwardRef(Link))
            className="makeStyles-articleLink-4"
            href="Article URL"
            target="_blank"
          >
            undefined
          </WithStyles(ForwardRef(Link))>
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-postTitle-9"
          variant="h5"
        >
          Post Title
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          align="justify"
          color="textSecondary"
          variant="subtitle1"
        >
          Post Description
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-username-8"
          color="textSecondary"
          variant="caption"
        >
          Bob Bobby
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
          variant="caption"
        >
          23 May
           -
           
          a few seconds ago
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(Card))>
    <ArticleInfo
      post={
        Object {
          "articleImageUrl": "Image URL",
          "articleUrl": "Article URL",
          "date": 1590216007250,
          "description": "Post Description",
          "isFavourited": false,
          "isPostLiked": false,
          "numberOfLikes": 2,
          "postId": 3028,
          "postType": "Culture",
          "title": "Post Title",
          "username": "Bob Bobby",
        }
      }
    />
  </WithStyles(ForwardRef(Card))>
</div>
`;
